### PR TITLE
fixed bugs from pe dry run

### LINKED ISSUE #109: [PE-D][Tester D] Repeated product titles
System does not recognise if products of the same name are added repeatedly. I think system should issue a warning stating that there is already a product with that name in the system in case the user forgot. 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: kirangeofran/ped#1

### LINKED ISSUE #134: [PE-D][Tester D] History PID number detection
When i want to check the history of PID 3 and i type the comand "history 3" it gives me the history of PID 2 instead.  Whenever you type in the PID number, it always gives you the history of the number before instead. 

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: kirangeofran/ped#6

### LINKED ISSUE #136: [PE-D][Tester D] History checking for invalide PID (continuation)
In the previous bug report, I stated how the history always shows the PID of the previous number instead of the PID you want to showcase. The UG also states that you should not be able to see the history of products that do not exist. In this case, when i type history 2,  I can see the history of PID 1 which does not exist anymore as seen from the list. 

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: kirangeofran/ped#7

### LINKED ISSUE #152: [PE-D][Tester E] Bug with History command
Not sure if I am interpreting the User guide wrongly but history does not show me the inflow and outflow for my product.

inputs:

1. inflow 4 a/5
2. inflow 4 a/10
3. history 4

actual outputs: 

1. Quantity updated. Quantity: 5
2. Quantity updated. Quantity: 15
3. No inflow or outflow found.

Expected outputs:
1. Quantity updated. Quantity: 5
2. Quantity updated. Quantity: 15
3. shows change in quantity and date of inflow etc. 

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: rexyyong/ped#7

### LINKED ISSUE #158: [PE-D][Tester C] HistoryCommand Sequence diagram incorrect format
TransactionList activation bar does not contain a smaller activation bar for the self-invocation of PastTransactions.

Class example:

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: john-nng/ped#5

### PR DESCRIPTION
**Pull request by EdmundTangg**
Update DG, resolves #158 (history puml)
Updates history command, resolves #134 and #136 (history command PID bug)
Updates add command and edit command, resolves #109 (repeated products bug)

### CODE REVIEW

File: src/main/java/seedu/stockpal/commands/HistoryCommand.java [hunk 1]
```diff
@@ -33,8 +33,8 @@ public HistoryCommand(Integer pidValue) {
 
     @Override
     public void execute(ProductList productList, TransactionList transactionList) throws StockPalException {
-        int productIndex = productList.findProductIndex(pid);
-        TransactionList.findTransactions(transactionList, productIndex + 1);
+        productList.findProductIndex(pid);
```
Comment (NgYaoDong) [inline src/main/java/seedu/stockpal/commands/HistoryCommand.java:36]: Would this line still be necessary?

File: src/main/java/seedu/stockpal/commands/EditCommand.java [hunk 2]
```diff
@@ -78,6 +78,13 @@ public void execute(ProductList productList) throws StockPalException {
         }
         int productIndex = productList.findProductIndex(this.pid);
         assert productList.getSize() > 0;
+
+        boolean repeated = productList.checkForRepeated(productList, name.getName());
+
+        if (repeated) {
+            throw new StockPalException("Duplicate item.");
```
Comment (NgYaoDong) [inline src/main/java/seedu/stockpal/commands/EditCommand.java]: Might want to make the string as a utility message

**Comment by EdmundTangg** [comment]
Resolves #152 as well.

File: src/main/java/seedu/stockpal/commands/HistoryCommand.java (hunk 1 above)
Comment (EdmundTangg) [reply src/main/java/seedu/stockpal/commands/HistoryCommand.java:36]: I would have to check whether PID is valid. Even though I do not need the index, this line allows me to check PID validity.

File: docs/images/FindCommandExecuteSequenceDiagram.png
Comment (Kobot7) [inline docs/images/FindCommandExecuteSequenceDiagram.png:1]: Perhaps add action bar for Ui

Comment (Kobot7) [inline docs/images/FindCommandExecuteSequenceDiagram.png:1]: Should ProductList action bar end right as it calls UI?

**Comment by EdmundTangg** [comment]
Resolves #136, 
#152 
#126

**Comment by EdmundTangg** [comment]
Resolves #128 as well.

**Comment by EdmundTangg** [comment]
Resolves #157 
#156

**Pull request closed without merging**